### DISCUSSION TITLE
windows version - ffmpeg record not writing correctly

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#282] [Questions]
## Which version are you using?

v0.13.1

## Which operating system are you using?

OS
- [X ] Windows

## Describe the issue

When getting inbound rtsp stream we are calling onpublish to record the stream to segments using this command:

runOnPublish: ffmpeg.exe -rtsp_transport tcp -i rtsp://localhost:$RTSP_PORT/$RTSP_PATH -vcodec copy -acodec copy -f segment -strftime 1 -segment_time 60 -segment_format mp4 ch3out%Y-%m-%d_%H-%M-%S.mp4

ffmpeg.exe is sitting in the same directory as the rtsp-simple-server.exe for testing. 

The files are created properly and ffmpeg runs. But the videos will not play. 

If i run the exact same command OUTSIDE of rtsp-simple-server, it works fine and file plays perfectly. 

## Describe how to replicate the issue
Replicate:
Push to the server, run onpublish command from above.
It saves the files and does everything, but the files won't play.

Do the same thing , but run the ffmpeg command (from the same folder as ffmpeg.exe and simpleserver) outside of rtsp simple server, it all works.  (Of course change rtsp_port/path to the correct info).  

for example:
ffmpeg.exe -rtsp_transport tcp -i rtsp://localhost:8554/ch3 -vcodec copy -acodec copy -f segment -strftime 1 -segment_time 60 -segment_format mp4 ch3out%Y-%m-%d_%H-%M-%S.mp4

**Comment by aler9** [comment]
Hello, i have various hypothesis about this.

First of all, try adding the `-nostdin` and `-y` flags to the ffmpeg command. In my personal scripts, i always use `-nostdin` and it allows to solve a lot of freezes and similar issues.

Second, i need to understand what you mean with "the videos won't play"
* they don't play in VLC? if that's the case, start vlc with the `-vvv` flag (verbose output) and report the output - maybe the SPS/PPS/iframes are missing and must be generated
* ffmpeg is stuck and doesn't go on? if that's the case, launch ffmpeg with the -loglevel verbose and report the output - maybe it could be a network problem

[a comment by an automated account is not shown]
